python/PyQt6/core/auto_generated/qgsattributes.sip.in

```diff
@@ -81,6 +81,22 @@ typedef QVector<QVariant> QgsAttributes;
     {
       qv->append( QVariant( QVariant::Int ) );
     }
+    else if ( PyBool_Check( obj ) )
+    {
+      qv->append( QVariant( PyObject_IsTrue( obj ) ) );
```

Comment:
https://docs.python.org/3/c-api/object.html#c.PyObject_IsTrue returns an integer value, but we need a bool here:

```suggestion
      qv->append( QVariant( PyObject_IsTrue( obj ) == 1 ) );
```

(Guess there's some test gaps here!)

Reply:
Good catch. Fixed.